integration-tests/src/main/java/io/quarkiverse/ironjacamar/ArtemisResourceAdapterFactory.java

```diff
@@ -1,64 +1,56 @@
 package io.quarkiverse.ironjacamar;
 
 import java.lang.reflect.Method;
+import java.util.Map;
 
-import jakarta.enterprise.context.ApplicationScoped;
-import jakarta.enterprise.inject.Produces;
-import jakarta.inject.Singleton;
-import jakarta.jms.ConnectionFactory;
 import jakarta.jms.Message;
 import jakarta.jms.MessageListener;
 import jakarta.resource.ResourceException;
 import jakarta.resource.spi.ActivationSpec;
-import jakarta.resource.spi.ConnectionManager;
+import jakarta.resource.spi.ManagedConnectionFactory;
 import jakarta.resource.spi.ResourceAdapter;
 import jakarta.resource.spi.endpoint.MessageEndpoint;
 
 import org.apache.activemq.artemis.core.remoting.impl.netty.NettyConnectorFactory;
-import org.apache.activemq.artemis.ra.ActiveMQRAConnectionFactory;
 import org.apache.activemq.artemis.ra.ActiveMQRAManagedConnectionFactory;
 import org.apache.activemq.artemis.ra.ActiveMQResourceAdapter;
 import org.apache.activemq.artemis.ra.inflow.ActiveMQActivationSpec;
 
-@Singleton
-public class AppResourceAdaptorSupport implements ResourceAdapterSupport {
+/**
+ * This would be in the artemis-jms extension
+ */
+@ResourceAdapterKind(value = ArtemisResourceAdapterFactory.KIND)
+@ResourceAdapterTypes(connectionFactoryTypes = { jakarta.jms.ConnectionFactory.class })
+public class ArtemisResourceAdapterFactory implements ResourceAdapterFactory {
```

Comment:
`ResourceAdapterSupport` is now `ResourceAdapterFactory` and requires two annotations to help the ironjacamar extension.

- The `@ResourceAdapterKind` is the value you use when specifying the `quarkus.ironjacamar.ra.kind` config
- The `@ResourceAdapterTypes` is necessary to allow `ConnectionFactory` objects to be injected in the app code